hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/ObjectEndpoint.java

```diff
@@ -1004,11 +1005,20 @@ void copy(OzoneVolume volume, InputStream src, long srcKeyLen,
       ReplicationConfig replication,
             Map<String, String> metadata) throws IOException {
     long copyLength;
-    try (OzoneOutputStream dest =
-                 getClientProtocol().createKey(
-        volume.getName(), destBucket, destKey, srcKeyLen,
-        replication, metadata)) {
-      copyLength = IOUtils.copyLarge(src, dest);
+    if (datastreamEnabled && !((replication != null &&
```

Comment:
```suggestion
    if (datastreamEnabled && !(replication != null &&
        replication.getReplicationType() == EC) &&
        srcKeyLen > datastreamMinLength) {
```

Please remove redundant parentheses.